eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Field.php

```diff
@@ -191,8 +174,95 @@ public function applyJoin(SelectQuery $query, SortClause $sortClause, $number)
                         $this->dbHandler->quoteColumn('version', $table),
                         $this->dbHandler->quoteColumn('current_version', 'ezcontentobject')
                     ),
-                    $languageExpression
+                    $this->getFieldCondition($query, $languageSettings, $table)
                 )
             );
     }
+
+    /**
+     * Returns a field language join condition for the given $languageSettings.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param array $languageSettings
+     * @param string $fieldTableName
+     *
+     * @return string
+     */
+    protected function getFieldCondition(SelectQuery $query, array $languageSettings, $fieldTableName)
```

Comment:
TBH I started with that approach, but it was hard to debug while I didn't know if the re-implementation of the filter is correct, so I changed to this one as it seemed easier. Now that https://github.com/ezsystems/ezpublish-kernel/pull/1428 is fixed and we have some tests in place, I can give it another try.